src/main/scala/is/hail/annotations/UnsafeRow.scala

```diff
@@ -254,15 +354,50 @@ class UnsafeRow(var ttBc: Broadcast[TypeTree],
 
   override def write(kryo: Kryo, output: Output) {
     kryo.writeObject(output, ttBc)
-    val enc = new Encoder(new DataOutputStream(output))
+    val enc = new Encoder()
     enc.writeRegionValue(ttBc.value.typ, region, offset)
+    kryo.writeObject(output, enc.outOff)
+    kryo.writeObject(output, enc.outMem)
+  }
+
+  private def writeObject(out: ObjectOutputStream) {
```

Comment:
How was serialization working before? maybe we can keep this separate as well to bring down the size of this PR?